dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -4897,18 +4897,35 @@ public Node visitWhileLoop(WhileLoopTree tree, Void p) {
 
             // Condition
             addLabelForNextNode(conditionStart);
+
+            // Is condition constant true?
+            boolean isWhileTrue = false;
+
             if (tree.getCondition() != null) {
+                Type condType = (Type) TreeUtils.typeOf(tree.getCondition());
+                isWhileTrue = condType.isTrue();
+
                 unbox(scan(tree.getCondition(), p));
-                ConditionalJump cjump = new ConditionalJump(loopEntry, loopExit);
-                extendWithExtendedNode(cjump);
+
+                if (!isWhileTrue) {
+                    ConditionalJump cjump = new ConditionalJump(loopEntry, loopExit);
+                    extendWithExtendedNode(cjump);
+                }
             }
 
             // Loop body
             addLabelForNextNode(loopEntry);
             if (tree.getStatement() != null) {
                 scan(tree.getStatement(), p);
             }
-            extendWithExtendedNode(new UnconditionalJump(conditionStart));
+
+            // If the condition is constant true, the successor of loop end is loop entry, instead
```

Comment:
What is the effect of this change? Is it really necessary to skip the condition for a while true?

Reply:
In fact, Javac bytecode generator merges the control flow at the end of the loop body with the condition start, so this change is just an optimization.